### PR TITLE
2531 make pghistory work on first changes to models

### PR DESCRIPTION
**Pull request by quevon24**
This PR contains the changes requested in #2531. This contains a few changes:

- A custom model tracker to store OLD value instead of NEW when making a change to an object, the custom tracker will also allow us to avoid duplicates when only updating the auto_now fields(save without making any changes).
- Replace Snapshot tracker with CustomSnapshot tracker
- Django migrations and sql files

### CODE REVIEW

**Comment by quevon24** [comment]
@mlissner The changes are complete, @albertisfu helped me to fix the tests related to #2591

File: cl/lib/pghistory.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import pgtrigger
+from pghistory import trigger
+from pghistory.core import DatabaseTracker, _get_name_from_label
+
+
+class CustomSnapshot(DatabaseTracker):
```
Comment (mlissner) [inline cl/lib/pghistory.py:6]: Perhaps:

```suggestion
class AfterUpdateOrDeleteSnapshot(DatabaseTracker):
```

Calling something "custom" doesn't tell the next person much about why it exists. :)

**Review by mlissner — CHANGES_REQUESTED** [review]
This is good stuff, Kevin. Two thoughts:

1. We should try to get this upstream instead of managing it ourselves. To that end, I updated the OLD vs. NEW bug in pghistory to ask if this would be a welcome feature, and I created a new issue to ask if `ignore_auto_now` would be a welcome feature: https://github.com/Opus10/django-pghistory/issues/83

    If we can get this upstreamed, we'll have the benefit of other people looking at your code, and we won't have to maintain it ourselves forever. Plus: Others might find it useful! 

2. The `Snapshot` class saves events when things are INSERTed or UPDATEd, right? But we want ours to do UPDATE or DELETE. I don't think I saw the events for delete though, right? We should add a test for this too.

I think that's it. Maybe let's fix number 2 now and we can proceed to merge this regardless of what happens with number 1. If we get your code into pghistory, we can eventually swap out our implementation for the one in the package.

Thank you!

File: cl/lib/pghistory.py (hunk 1 above)
Comment (quevon24) [reply cl/lib/pghistory.py:6]: @mlissner Yes, the Snapshot class works for inserts and updates, we just talked about the behavior when we do an update, we don't have triggers for delete action, i can add them, i think we should also store the old value when we do a deletion, right?

Comment (mlissner) [reply cl/lib/pghistory.py:6]: Yep, exactly.

**Comment by quevon24** [comment]
@mlissner, all changes are complete, i added the "ignore_field_changes" argument to exclude fields from row comparison, but not from copy

**Comment by quevon24** [comment]
@mlissner the changes are ready with the new version of pghistory, 

i added by default the option to exclude all auto_now fields in a custom snapshot class instead of using the new barebone tracker "BeforeUpdateOrDelete" to avoid those extra records in the db caused by auto_now fields, this is ok, or should we use the barebone tracker?

File: cl/people_db/models.py [hunk 2]
```diff
@@ -60,7 +61,7 @@
 
 
 @pghistory.track(
-    pghistory.Snapshot(),
+    AfterUpdateOrDeleteSnapshot(),
```
Comment (mlissner) [inline cl/people_db/models.py:64]: Can you make all of these into one line, instead of three?

File: cl/search/models.py [hunk 3]
```diff
@@ -81,7 +82,7 @@ class PRECEDENTIAL_STATUS:
 
 
 @pghistory.track(
-    pghistory.Snapshot(),
+    AfterUpdateOrDeleteSnapshot(),
```
Comment (mlissner) [inline cl/search/models.py:85]: More triple-line issues here. I don't usually complain, but there are so many of them! :)

**Review by mlissner — CHANGES_REQUESTED** [review]
This looks great. Just a couple little comments, and I think we'll be ready to merge this monster. Thank you.

**Comment by mlissner** [comment]
Oh, sorry, and it looks like poetry is unhappy in the merge. Revert the file and add the dependency fresh?

**Comment by quevon24** [comment]
@mlissner changes are ready

**Review by mlissner — APPROVED** [review]

**Comment by mlissner** [comment]
Fantastic. Here we go, thank you!

**Pull request merged**